### PR TITLE
#555 update test names

### LINKED ISSUE #555: Tests-Naming Convention für AuthService
Wie in der [Doku](https://it-at-m.github.io/Wahllokalsystem/technik/coding_conventions/tests_naming.html) festgehalten, wollen wir die Namen der Tests des Services nach dem Schema `should_<result>_when_<input>` umformulieren.

- Beispiele siehe [Vorfälle und Vorkommnisse Service](https://github.com/it-at-m/Wahllokalsystem/tree/dev/wls-vorfaelleundvorkommnisse-service/src/test/java/de/muenchen/oss/wahllokalsystem/vorfaelleundvorkommnisseservice)
- konkretes Beispiel für den [Aufbau von Tests für overload methods](https://github.com/it-at-m/Wahllokalsystem/blob/8fcc8f5639627bdba25dc53b696ea2fe7826c6f8/wls-vorfaelleundvorkommnisse-service/src/test/java/de/muenchen/oss/wahllokalsystem/vorfaelleundvorkommnisseservice/service/ereignis/EreignisModelMapperTest.java#L54-L92)

Die sich wiederholenden Tests im `.configuration`-Package wurden bereits im Broadcast-Service umbenannt. Bei der neuen Benennung kann sich [hier](https://github.com/it-at-m/Wahllokalsystem/tree/dev/wls-broadcast-service/src/test/java/de/muenchen/oss/wahllokalsystem/broadcastservice/configuration) daran orientiert werden.

Akzeptanzkriterien:
- [ ] alle Testnamen sind auf englisch
- [ ] alle Testnamen folgen dem Muster `should_<result>_when_<input>` 
- [ ] `result` und `input` sind in CamelCase gehalten
- [ ] die `@Nested` class bekommt den Namen der zu testenden Methode
- [ ] method overloads werden durch zusätliche `@Nested`-Blöcke innerhalb gruppiert

### PR DESCRIPTION
**Pull request by vjohnslhm**
# Beschreibung:

Alle Tests wurden entsprechend der Naming Convention umbenannt.

### Backend ###
- [ ] alle tests laufen durch
- [ ] alle tests korrekt benannt

# Referenzen[^1]:

Verwandt mit Issue #

Closes #555

> [^1]: _Nicht zutreffende Referenzen vor dem Speichern entfernen_

## Summary by CodeRabbit

- **Refactor**
	- Improved naming conventions for test methods to enhance clarity and readability.
	- Standardized formatting in the test code for consistency.
	- Introduced new unit tests for the `BaseEntityWrapper` class to validate hash code and equality behaviors.

### CODE REVIEW

**Comment by vjohnslhm** [comment]
@coderabbitai full review

**Comment by vjohnslhm** [comment]
@coderabbitai review

**Comment by vjohnslhm** [comment]
@coderabbitai bitte teste alle weiteren files aus der `wls-auth-service/src/test` directory auf die einhaltung der naming convention. zähle alle tests auf, die noch umbenannt werden müssen.

**Comment by vjohnslhm** [comment]
@coderabbitai review

File: wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/configuration/nfcconverter/NfcHelperTest.java [hunk 1]
```diff
@@ -34,7 +36,7 @@ class NfcHelperTest {
     private static final String[] NFC_OUTPUT_EXPECTED = new String[] { FIRST_NFC, SECOND_NFC, THIRD_NFC };
 
     @Test
-    void nfcConverterString() {
+    void should_wrongNameOnPurpose() {
```
[the remark this answers is not in the document]
Comment (vjohnslhm) [reply wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/configuration/nfcconverter/NfcHelperTest.java]: updated with [49c7015](https://github.com/it-at-m/Wahllokalsystem/pull/602/commits/49c70159f4e2bf0e0874ec84dadd002953fa8912)

File: wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/configuration/nfcconverter/NfcHelperTest.java [hunk 2]
```diff
@@ -48,7 +50,7 @@ void nfcConverterString() {
     }
 
     @Test
-    void nfcConverterStringBuffer() {
+    void should_beDetected_fromCodeRabbit_when_doingItsReview() {
```
[the remark this answers is not in the document]
Comment (vjohnslhm) [reply wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/configuration/nfcconverter/NfcHelperTest.java]: updated with [49c7015](https://github.com/it-at-m/Wahllokalsystem/pull/602/commits/49c70159f4e2bf0e0874ec84dadd002953fa8912)

File: wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/domain/CodeRabbitTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+class CodeRabbitTest {
+
+    private static final String UUID_STRING_1 = "8b87b673-1eb0-480f-a0e6-622ef860f790";
+    private static final String UUID_STRING_2 = "f804b2b0-d259-4a7d-b246-156ea545dc63";
+
+    @Nested
+    class HashCode {
+
+        @Test
+        void newTestForCodeRabbitToDetect() {
```
[the remark this answers is not in the document]
Comment (vjohnslhm) [reply wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/domain/CodeRabbitTest.java:21]: updated with [49c7015](https://github.com/it-at-m/Wahllokalsystem/pull/602/commits/49c70159f4e2bf0e0874ec84dadd002953fa8912)

[the remark this answers is not in the document]
Comment (vjohnslhm) [reply wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/domain/CodeRabbitTest.java]: updated with [8ff1e29](https://github.com/it-at-m/Wahllokalsystem/pull/602/commits/8ff1e29016888a17b92ab243d7f24f1772da6ecd)

File: wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/domain/CodeRabbitTest.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+            Assertions.assertThat(baseEntity1.hashCode()).isEqualTo(baseEntity2.hashCode());
+        }
+    }
+
+    @Nested
+    class Equals {
+
+        @Test
+        void should_notBeDetected_when_codeRabbitReviews() {
```
[the remark this answers is not in the document]
Comment (vjohnslhm) [reply wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/domain/CodeRabbitTest.java]: updated with [8ff1e29](https://github.com/it-at-m/Wahllokalsystem/pull/602/commits/8ff1e29016888a17b92ab243d7f24f1772da6ecd)

**Comment by vjohnslhm** [comment]
@coderabbitai bitte teste nochmal alle files aus der wls-auth-service/src/test directory auf die einhaltung der naming convention. zähle alle tests auf, die noch umbenannt werden müssen.

File: wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/configuration/UnicodeConfigurationTest.java [hunk 5]
```diff
@@ -12,8 +12,10 @@
 import de.muenchen.oss.wahllokalsystem.authservice.MicroServiceApplication;
 import de.muenchen.oss.wahllokalsystem.authservice.domain.TheEntity;
 import de.muenchen.oss.wahllokalsystem.authservice.rest.TheEntityRepository;
+
 import java.net.URI;
 import java.util.UUID;
+
 import org.apache.commons.lang3.StringUtils;
```
Comment (MrSebastian) [inline wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/configuration/UnicodeConfigurationTest.java]:
```suggestion
import de.muenchen.oss.wahllokalsystem.authservice.domain.TheEntity;
import de.muenchen.oss.wahllokalsystem.authservice.rest.TheEntityRepository;
import java.net.URI;
import java.util.UUID;
import org.apache.commons.lang3.StringUtils;
```

File: wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/configuration/nfcconverter/NfcConverterTest.java [hunk 6]
```diff
@@ -14,6 +14,7 @@
 import jakarta.servlet.http.HttpServletRequest;
 import jakarta.servlet.http.HttpServletResponse;
 import jakarta.servlet.http.Part;
+
 import java.io.BufferedReader;
```
Comment (MrSebastian) [inline wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/configuration/nfcconverter/NfcConverterTest.java]:
```suggestion
import jakarta.servlet.http.Part;
import java.io.BufferedReader;
```

File: wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/configuration/nfcconverter/NfcConverterTest.java [hunk 7]
```diff
@@ -27,6 +28,7 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+
 import org.apache.commons.collections4.list.UnmodifiableList;
```
Comment (MrSebastian) [inline wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/configuration/nfcconverter/NfcConverterTest.java]:
```suggestion
import java.util.Map;
import org.apache.commons.collections4.list.UnmodifiableList;
```

File: wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/configuration/nfcconverter/NfcHelperTest.java [hunk 8]
```diff
@@ -9,10 +9,12 @@
 import static org.junit.jupiter.api.Assertions.assertNull;
 
 import jakarta.servlet.http.Cookie;
+
 import java.util.Arrays;
```
Comment (MrSebastian) [inline wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/configuration/nfcconverter/NfcHelperTest.java]:
```suggestion
import jakarta.servlet.http.Cookie;
import java.util.Arrays;
```

File: wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/configuration/nfcconverter/NfcHelperTest.java [hunk 9]
```diff
@@ -9,10 +9,12 @@
...
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
+
 import org.junit.jupiter.api.Test;
```
Comment (MrSebastian) [inline wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/configuration/nfcconverter/NfcHelperTest.java]:
```suggestion
import java.util.Map;
import org.junit.jupiter.api.Test;
```

File: wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/domain/UserRepositoryImplIntegrationTest.java [hunk 10]
```diff
@@ -4,9 +4,11 @@
 import de.muenchen.oss.wahllokalsystem.authservice.TestConstants;
 import de.muenchen.oss.wahllokalsystem.authservice.configuration.Profiles;
 import jakarta.persistence.EntityManager;
+
 import java.util.Collections;
```
Comment (MrSebastian) [inline wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/domain/UserRepositoryImplIntegrationTest.java]:
```suggestion
import jakarta.persistence.EntityManager;
import java.util.Collections;
```

File: wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/domain/UserRepositoryImplIntegrationTest.java [hunk 11]
```diff
@@ -4,9 +4,11 @@
...
 import java.util.List;
 import java.util.Set;
+
 import lombok.val;
```
Comment (MrSebastian) [inline wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/domain/UserRepositoryImplIntegrationTest.java]:
```suggestion
import java.util.Set;
import lombok.val;
```

File: wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/domain/UserRepositoryImplTest.java [hunk 12]
```diff
@@ -4,12 +4,14 @@
 import static org.mockito.Mockito.times;
 
 import de.muenchen.oss.wahllokalsystem.authservice.service.CryptoService;
+
 import java.util.Collections;
```
Comment (MrSebastian) [inline wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/domain/UserRepositoryImplTest.java]:
```suggestion
import de.muenchen.oss.wahllokalsystem.authservice.service.CryptoService;
import java.util.Collections;
```

File: wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/domain/UserRepositoryImplTest.java [hunk 13]
```diff
@@ -4,12 +4,14 @@
...
 import java.util.LinkedList;
 import java.util.List;
 import java.util.Optional;
 import java.util.UUID;
 import java.util.stream.StreamSupport;
+
 import lombok.val;
```
Comment (MrSebastian) [inline wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/domain/UserRepositoryImplTest.java]:
```suggestion
import java.util.stream.StreamSupport;
import lombok.val;
```

File: wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/rest/WahllokalBenutzerControllerIntegrationTest.java [hunk 14]
```diff
@@ -14,9 +14,11 @@
 import de.muenchen.oss.wahllokalsystem.authservice.domain.UserRepository;
 import de.muenchen.oss.wahllokalsystem.authservice.service.CryptoService;
 import jakarta.persistence.EntityManager;
+
 import java.time.LocalDate;
```
Comment (MrSebastian) [inline wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/rest/WahllokalBenutzerControllerIntegrationTest.java]:
```suggestion
import jakarta.persistence.EntityManager;
import java.time.LocalDate;
```

File: wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/rest/WahllokalBenutzerControllerIntegrationTest.java [hunk 15]
```diff
@@ -14,9 +14,11 @@
...
 import java.util.Collections;
 import java.util.List;
+
 import lombok.val;
```
Comment (MrSebastian) [inline wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/rest/WahllokalBenutzerControllerIntegrationTest.java]:
```suggestion
import java.util.List;
import lombok.val;
```

**Review by MrSebastian — CHANGES_REQUESTED** [review]

File: wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/rest/WahllokalBenutzerControllerIntegrationTest.java (hunk 15 above)
Comment (vjohnslhm) [reply wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/rest/WahllokalBenutzerControllerIntegrationTest.java]: -> commit [aa49f99](https://github.com/it-at-m/Wahllokalsystem/pull/602/commits/aa49f99ee100728e2ab9f3da233e9cc8f073309d)

File: wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/rest/WahllokalBenutzerControllerIntegrationTest.java (hunk 14 above)
Comment (vjohnslhm) [reply wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/rest/WahllokalBenutzerControllerIntegrationTest.java]: -> commit [aa49f99](https://github.com/it-at-m/Wahllokalsystem/pull/602/commits/aa49f99ee100728e2ab9f3da233e9cc8f073309d)

File: wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/domain/UserRepositoryImplTest.java (hunk 13 above)
Comment (vjohnslhm) [reply wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/domain/UserRepositoryImplTest.java]: -> commit [aa49f99](https://github.com/it-at-m/Wahllokalsystem/pull/602/commits/aa49f99ee100728e2ab9f3da233e9cc8f073309d)

File: wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/domain/UserRepositoryImplTest.java (hunk 12 above)
Comment (vjohnslhm) [reply wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/domain/UserRepositoryImplTest.java]: -> commit [aa49f99](https://github.com/it-at-m/Wahllokalsystem/pull/602/commits/aa49f99ee100728e2ab9f3da233e9cc8f073309d)

File: wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/domain/UserRepositoryImplIntegrationTest.java (hunk 11 above)
Comment (vjohnslhm) [reply wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/domain/UserRepositoryImplIntegrationTest.java]: -> commit [aa49f99](https://github.com/it-at-m/Wahllokalsystem/pull/602/commits/aa49f99ee100728e2ab9f3da233e9cc8f073309d)

File: wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/domain/UserRepositoryImplIntegrationTest.java (hunk 10 above)
Comment (vjohnslhm) [reply wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/domain/UserRepositoryImplIntegrationTest.java]: -> commit [aa49f99](https://github.com/it-at-m/Wahllokalsystem/pull/602/commits/aa49f99ee100728e2ab9f3da233e9cc8f073309d)

File: wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/configuration/nfcconverter/NfcHelperTest.java (hunk 9 above)
Comment (vjohnslhm) [reply wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/configuration/nfcconverter/NfcHelperTest.java]: -> commit [aa49f99](https://github.com/it-at-m/Wahllokalsystem/pull/602/commits/aa49f99ee100728e2ab9f3da233e9cc8f073309d)

File: wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/configuration/nfcconverter/NfcHelperTest.java (hunk 8 above)
Comment (vjohnslhm) [reply wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/configuration/nfcconverter/NfcHelperTest.java]: -> commit [aa49f99](https://github.com/it-at-m/Wahllokalsystem/pull/602/commits/aa49f99ee100728e2ab9f3da233e9cc8f073309d)

File: wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/configuration/nfcconverter/NfcConverterTest.java (hunk 7 above)
Comment (vjohnslhm) [reply wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/configuration/nfcconverter/NfcConverterTest.java]: -> commit [aa49f99](https://github.com/it-at-m/Wahllokalsystem/pull/602/commits/aa49f99ee100728e2ab9f3da233e9cc8f073309d)

File: wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/configuration/nfcconverter/NfcConverterTest.java (hunk 6 above)
Comment (vjohnslhm) [reply wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/configuration/nfcconverter/NfcConverterTest.java]: -> commit [aa49f99](https://github.com/it-at-m/Wahllokalsystem/pull/602/commits/aa49f99ee100728e2ab9f3da233e9cc8f073309d)

File: wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/configuration/UnicodeConfigurationTest.java (hunk 5 above)
Comment (vjohnslhm) [reply wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/configuration/UnicodeConfigurationTest.java]: -> commit [aa49f99](https://github.com/it-at-m/Wahllokalsystem/pull/602/commits/aa49f99ee100728e2ab9f3da233e9cc8f073309d)

**Review by MrSebastian — DISMISSED** [review]

**Review by MrSebastian — APPROVED** [review]

**Pull request merged**